include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+            return false;
+        }
+
+        // Set Options for what events we get from Google
+        $optParams = array(
+            'maxResults' => 500,
+            //'orderBy' => 'startTime',
+            'showDeleted' => true,
+            'singleEvents' => true,
+            'timeMin' => date('c', strtotime('-1 month')),
+            'timeMax' => date('c', strtotime('+3 month')),
+        );
+
+        $results_g = $this->gService->events->listEvents($this->calendarId, $optParams);
+
+        // We only want the events, not the leading cruft
+        $results = $results_g->getItems();
+
+        if (empty($results))
+        {
+            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'No events found.');
+            return Array();
+        }
+        else {
+            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Found ' . count($results) . ' Google Events');
+            return $results;
+        }
+    }
+
+    /**
+     * Get a google event by the matching SuiteCRM meeting id
+     *
+     * @param string $meeting_id
+     *
+     * @return Google_Service_Calendar_Event
+     */
+    public function getGoogleEventByMeetingId($meeting_id)
+    {
+
+        // Make sure the calendar service is set up
+        if (!isset($this->gService))
+        {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
+            return false;
+        }
+
+        // Set Options for what events we get from Google
+        $optParams = array(
+            'maxResults' => 2, // We ask for a max of 2 because we should *NEVER* get more than 1. If we get 2, something is horribly wrong.
+            'showDeleted' => true,
+            'singleEvents' => false,
+            'timeMin' => date('c', strtotime('-6 month')),
+            'privateExtendedProperty' => 'suitecrm_id=' . $meeting_id
+        );
+
+        $results_g = $this->gService->events->listEvents($this->calendarId, $optParams);
+        $results = $results_g->getItems();
+
+        if (count($results) > 1)
+        {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'We received more than one Google event with the same SuiteCRM meeting ID. Something is horribly wrong!');
+            return false;
+        }
+        elseif (count($results) == 0) { // No events match. Return emtpy array.
+            return Array();
+        }
+        else {
+            $gEvent = $results[0];
+            return $gEvent;
+        }
+    }
+
+    /**
+     * Get a google event by the event id
+     *
+     * @param string $event_id Google Event ID
+     *
+     * @return Google_Service_Calendar_Event|null Google_Service_Calendar_Event if found, null if not found
```

Comment:
This method returns void instead of null. Please update doc block to match.

Reply:
It should return null. My understanding is that with php void and null are the same thing. Is that incorrect? In any case, I'm setting it explicitly to null so it's consistent.